shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppActionDispatcher.kt

```diff
@@ -50,8 +50,18 @@ class AppActionDispatcher(
 
                 sentryInteractor.addBreadcrumb(HyperskillSentryBreadcrumbBuilder.buildAppDetermineUserAccountStatus())
 
-                profileInteractor
-                    .getCurrentProfile(sourceType = DataSourceType.REMOTE)
+                val isAuthorized = authInteractor.isAuthorized()
+                    .getOrDefault(false)
+
+                val profileResult = if (isAuthorized) {
+                    profileInteractor
+                        .getCurrentProfile(sourceType = DataSourceType.CACHE)
+                        .onFailure { profileInteractor.getCurrentProfile(sourceType = DataSourceType.REMOTE) }
+                } else {
+                    profileInteractor.getCurrentProfile(sourceType = DataSourceType.REMOTE)
```

Comment:
Why should we fetch a profile even if the user is not authorized?

Reply:
This is necessary for the correct working of analytics:

- When the user is unauthorized we perform POST `/api/frontend-events` with cookies headers -> `/api/profiles/current` returns `set-cookie` header
- When the user is authorized we perform POST `/api/frontend-events` with a bearer token

Reply:
To skip profile loading when a user is unauthorized we need to implement caching for `HttpCookiesPlugin`